### ISSUE TITLE
Backup out of phone

### ISSUE THREAD

**Issue by DJaeger** [jensstein/oandbackup#90]
It should be possible to backup to an network-share or cloud-solution.
It makes it possible to safe your space on phone and have your data available if your phone get demaged, lost or stolen.
I think this is the main intend for backing up your phone-data.

So it could be possible to sheduled backup your data, every night if you are in you home-network and your phone is loading, to an network-share or for example to your owncloud.

**Comment by jensstein** [comment]
almost all of the widely used cloud providers use proprietary software, so i wouldn't want to include support for something like dropbox.
support for things like owncloud or syncthing might be nice, but it would probably be better just to set that particular cloud-provider client to syncronize the backup directory, wouldn't it? 
otherwise i'm afraid there would only be support for one or two cloud providers that not very many people actually use.

**Comment by DJaeger** [comment]
A lot of them use WebDAV.
Offer the possibility to upload to WebDAV and let it open to the user which sollution they use.
In addition you could offer an dropdown, in which they can choose their solution to prefill the uploadpath-field.

Some largely german examples:
Own configuration:
Empty
1&1 Online-Storage:
https://webdav.office.1und1.de/
Box:
https://dav.box.com/dav/
CloudMe:
https://webdav.cloudme.com/USERNAME/CloudDrive/Documents/
DriveOnWeb:
https://storage.driveonweb.de/probdav/
GMX MediaCenter:
https://webdav.mc.gmx.net/
MyDrive:
https://webdav.mydrive.ch/
ownCloud:
https://YOURSERVER/owncloud/remote.php/webdav/
Strato HiDrive Media:
https://webdav.hidrive.strato.com/users/USERNAME/
TelekomCloud Mediacenter:
https://webdav.mediencenter.t-online.de/
Trend Micro SafeSync:
https://dav.safesync.com/
WEB.DE Online-Storage:
https://webdav.smartdrive.web.de/

**Comment by jensstein** [comment]
okay, i hadn't thought of webdav for this. this might be a good idea.
do you know if you just need credentials with these providers or would you also need api keys or something like that?

**Comment by DJaeger** [comment]
These Providers by default all work with only credentials.
These Providers are also used by an (closed source) App, that only have fields for provider (dropdown), URL (prefilled by provider choose), username, password and domain (for authentisation against ActiveDirectory for corporates or something like this - only active for own configuration and ownCloud)

**Comment by DJaeger** [comment]
I'm think about Seafile also, that have an WebDAV module, but I have not to looked for an default URL until now.

**Comment by DJaeger** [comment]
You could use Apache Jackrabbit:
https://jackrabbit.apache.org/

Reference-Implementation:
https://github.com/owncloud/android-library

**Comment by jensstein** [comment]
fine, i will look into this at some point then. thanks for the references.
i will probably not want to add another library dependence but i will see if the code can be useful just to make the upload functionality.

**Comment by DJaeger** [comment]
Any news on this?

**Comment by ildar** [comment]
Did you still consider syncing the backup folder with a cloud's own
software?
Then you would have a scheduled backup made by oandbackup and scheduled
sync to cloud made by the cloud's own app?
This was mentioned by Jens, but maybe unintendedly ignored?

**Comment by DJaeger** [comment]
The ownCloud app can only locally autoupdate in the cloud existing files and autoupload photos and videos.
Also I think it is much better to let the app itself take care of, so that it only syncs if there is something new as the app can simply update in the cloud like it would update locally.
I have not ignored it, but Jens said he would look into this.

**Comment by jensstein** [comment]
no, unfortunately i haven't had much time to work on the project at all in the last half year. so i'm afraid this will have to wait a little longer. but i'll keep you posted when there's any progess. and of course i'll happily take patches or other relevant pointers.

**Comment by DJaeger** [comment]
No problem.
I would give you more info on this or maybe a PR if I find some time for it.
